perfkitbenchmarker/flag_util.py

```diff
@@ -304,3 +306,60 @@ def DEFINE_units(name, default, help, convertible_to=None,
   serializer = UnitsSerializer()
 
   flags.DEFINE(parser, name, default, help, flag_values, serializer, **kwargs)
+
+
+# The YAML flag type is necessary because flags can be read either via
+# the command line or from a config file. If they come from a config
+# file, they will already be parsed as YAML, but if they come from the
+# command line, they will be raw strings. The point of this flag is to
+# guarantee a consistent representation to the rest of the program.
+class YAMLParser(flags.ArgumentParser):
+  """Parse a flag containing YAML."""
+
+  syntactic_help = 'A YAML expression. Ex: "{x: y, foo: 2.3}".'
```

Comment:
Actually, I recall gflags has validators that can be attached to specific flags if you want to go the instance member route.